src/domain/books/BooksPage.tsx

```diff
@@ -1,7 +1,45 @@
-export const BooksPage = (): JSX.Element => {
+import React, { useEffect, useState } from "react";
+import { BookCard } from "../../components/BookCard";
+import { getBooks } from "./booksSlice";
+import { useDispatch, useSelector } from "react-redux";
+import { AppDispatch, RootState } from "../../store/store";
+import type { Book } from "../../domain/books/types";
+import { Skeleton } from "antd";
+import "./style.css";
+
+const BooksPage = () => {
+  const dispatch: AppDispatch = useDispatch();
+  const Bookdata = useSelector((state: RootState) => state.books.books);
+  const [bookData, setBookData] = useState<Book[]>(Bookdata);
+  useEffect(() => {
+    dispatch(getBooks());
+  }, []);
+  useEffect(() => {
+    setBookData(Bookdata);
+  }, [Bookdata]);
```

Comment:
If we are storing a state in redux then it doesn't make any sense to copy it to local component.
Also, use the create the selectors inside the slice files and use them here.

`const books = useSelector(booksSelector)`

Reply:
yes they is no need to do that locally... I will change that